pkg/settlement/swap/transaction/transaction.go

```diff
@@ -241,3 +255,16 @@ func (t *transactionService) nextNonce(ctx context.Context) (uint64, error) {
 func (t *transactionService) putNonce(nonce uint64) error {
 	return t.store.Put(t.nonceKey(), nonce)
 }
+
+func (t *transactionService) GetRedoNonce() (uint64, error) {
```

Comment:
I would rather put the logic in the `nextNonce` method based on whether or `ctx` has a `redo` flag. Plus the `nextNonce` method is private so really why are you making this method public anyway?